service-front/module/Application/src/Controller/DonorFlowController.php

```diff
@@ -95,15 +97,57 @@ public function howWillDonorConfirmAction(): ViewModel|Response
                             $uuid,
                             $data
                         );
-                        return $this->redirect()->toRoute("root/donor_details_match_check", ['uuid' => $uuid]);
+                        $returnRoute = "root/donor_details_match_check";
                     }
+                    return $this->redirect()->toRoute($returnRoute, ['uuid' => $uuid]);
                 }
             }
         }
 
         return $view->setTemplate($templates['default']);
     }
 
+    public function whatIsVouchingAction(): ViewModel|Response
+    {
+        $view = new ViewModel();
+        $uuid = $this->params()->fromRoute("uuid");
+        $detailsData = $this->opgApiService->getDetailsData($uuid);
+        $view->setVariable('details_data', $detailsData);
+
+        if ($this->getRequest()->isPost()) {
+            $formData = $this->getRequest()->getPost()->toArray();
+            if ($formData['confirm_vouching'] == 'yes') {
+                /**
+                 * @psalm-suppress ArgumentTypeCoercion
+                 */
+                $pdf = $this->siriusApiService->sendPdf($detailsData, "DLP-VOUCH-INVITE", $this->request);
+                if ($pdf['status'] === 201) {
+                    return $this->redirect()->toRoute("root/vouching_what_happens_next", ['uuid' => $uuid]);
+                } else {
+                    // what should i do with this error??
+                    $view->setVariable('errors', ['API Error - Could not send letter']);
```

Comment:
Does the template you're using handle displaying errors? If so you can let the view rendering take care of it. If not then you may want to throw an exception that the generic handlers will pick up. Also worth logging that the error occured.

Reply:
Worth testing by breaking the mock (e.g. remove the 201 response from the OpenAPI file), but I think it will already throw an exception on failure and that will be caught by the framework to show an error page.

i.e. you don't need to add any handling for the error condition as this code will never be reached

Reply:
yep - tested removing the 201 response and you get sent to an error page and never see this error, will remove.